CONTRIBUTING.md

```diff
@@ -62,3 +110,23 @@ Thanks for taking the plunge!
   rather than `[1., 2.]`
 * In docstrings, optional arguments, including separators and spaces, are surrounded by brackets,
   e.g. `mymethod(required[, optional1[, optional2] ]; kwargs...)`
+
+## Git Recommendations For Pull Requests
+
+* Avoid working from the `main` branch of your fork, creating a new branch will make it easier if DataFrame's `main` branch changes and you need to update your pull request.
+
+* Try to [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) together small commits that make repeated changes to the same section of code so your pull request is easier to review, and DataFrame's won't have any broken intermediate commits. A reasonable number of separate well-factored commits is fine, especially for larger changes.
+
+* If any conflicts arise due to changes in DataFrame's `main` branch, prefer updating your pull request branch with `git rebase` versus `git merge` or `git pull`, since the latter will introduce merge commits that clutter the git history with noise that makes your changes more difficult to review.
+
+* Descriptive commit messages are good.
+
+* Using `git add -p` or `git add -i` can be useful to avoid accidently committing unrelated changes.
+
+* GitHub does not send notifications when you push a new commit to pull request, so please add a comment to the pull request thread to let reviewers know when you've made changes.
+
+* When linking to specific lines of code in discussion of an issue or pull request, hit the `y` key while viewing code on GitHub to reload the page with a URL that includes the specific version that you're viewing. That way any lines of code that you refer to wil still make sense in the future, even if the content of the file changes.
```

Comment:
I understand your intention here, but I am not sure I fully understand the recommendation. I.e. when I kit `y` key nothing happens.

Reply:
It happens when you select some code of any file and then hit the `y` key: let's say I select the first 4 lines of [core.jl](https://github.com/JuliaData/DataFrames.jl/tree/main/src/join) then after selection url will look like this: https://github.com/JuliaData/DataFrames.jl/blob/a671a3b86537a253d461ff54b87ae2c26e4e8a04/src/join/core.jl
and before selection:
https://github.com/JuliaData/DataFrames.jl/tree/main/src/join